src/components/VerticalTabs/VerticalTabs.tsx

```diff
@@ -0,0 +1,202 @@
+    `,
+};
+Title.propName = 'Title';
+
+/** Vertical Tabs Component */
+interface IVerticalTabsPropsRaw extends StandardProps {
+	/** Custom Tab component (alias for `VerticalTabs.Tab`) */
+	Tab?: React.ReactNode;
+
+	/** Custom Title component (alias for `VerticalTabs.Title`) */
+	Title?: React.ReactNode;
+
+	/** Indicates which of the \`VerticalTabs.Tab\` children is currently
+    selected */
+	selectedIndex?: number;
+
+	/** Callback fired when the user selects a \`VerticalListMenu.Item\`.*/
+	onSelect: (
+		index: number,
+		{
+			event,
+			props,
+		}: {
+			event: React.MouseEvent;
+			props: IVerticalListMenuItemProps;
+		}
+	) => void;
+}
+
+type IVerticalTabsProps = Overwrite<
+	React.DetailedHTMLProps<React.HTMLAttributes<HTMLDivElement>, HTMLDivElement>,
+	IVerticalTabsPropsRaw
+>;
+
+/** Default props for the VerticalTabs component */
+const defaultProps = {
+	selectedIndex: 0,
+	onSelect: _.noop,
+};
+
+export interface IVerticalTabsState {
+	selectedIndex: number;
+}
+
+class VerticalTabs extends React.Component<
+	IVerticalTabsProps,
+	IVerticalTabsState
+> {
+	static displayName = 'VerticalTabs';
+	static propTypes = {
+		className: string`
+			Class names that are appended to the defaults.
+		`,
+
+		selectedIndex: number`
+			Indicates which of the \`VerticalTabs.Tab\` children is currently
+			selected. The index of the last \`VerticalTabs.Tab\` child with
+			\`isSelected\` equal to \`true\` takes precedence over this prop.
+		`,
+
+		onSelect: func`
+			Callback for when the user clicks a tab. Called with the index of the tab
+			that was clicked.  Signature: \`(index, { event, props}) => {}\`
+		`,
+	};
+
+	static defaultProps = defaultProps;
+	static reducers = reducers;
+	static Tab = Tab;
+	static Title = Title;
+
+	// For backward compatibility with buildHybridComponent
+	static definition = {
+		statics: {
+			reducers,
+			Tab,
+			Title,
+			peek: {
+				description: `
+                \`VerticalTabs\` provides vertically tabbed navigation. It has a
+                flexible interface that allows tab content to be passed as regular
+                React children or through props.
+            `,
+				categories: ['navigation'],
+				madeFrom: ['VerticalListMenu'],
+			},
+		},
+	};
+
+	render(): React.ReactNode {
+		const { className, onSelect, selectedIndex, ...passThroughs } = this.props;
+
+		// Grab props array from each Tab
+		const tabChildProps = _.map(
+			findTypes(this.props, VerticalTabs.Tab),
+			'props'
+		);
+
+		const selectedIndexFromChildren = _.findLastIndex(tabChildProps, {
+			isSelected: true,
+		});
+
+		const actualSelectedIndex =
+			selectedIndexFromChildren !== -1
+				? selectedIndexFromChildren
+				: selectedIndex;
+
+		return (
+			<div
+				{...omitProps(passThroughs, undefined, _.keys(VerticalTabs.propTypes))}
+				className={cx('&', className)}
+			>
+				<VerticalListMenu
+					selectedIndices={[
+						_.isUndefined(actualSelectedIndex) ? 0 : actualSelectedIndex,
```

Comment:
An example I saw in the code base for a similar index had "" instead of a zero; the default is 0, and type is number, so a zero seemed more precise, even though "" is == a truthy zero (but not === 0)

Reply:
Isn't this check redundant with the default prop though?

Reply:
Good point. It is redundant. I removed it.